### PR TITLE
New package: obs-0.10.1

### PR DESCRIPTION
**Pull request by lemmi**
Open Broadcaster Software Alpha Release

### CODE REVIEW

File: srcpkgs/obs/template [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'obs'
+pkgname=obs
+version=0.10.1
+revision=1
+wrksrc=obs-studio-${version}
+only_for_archs="i686 x86_64"
```
Comment (xtraeme) [inline srcpkgs/obs/template]: Please add `i686-musl x86_64-musl` too.

**Comment by lemmi** [comment]
@xtraeme musl build fails because pthread_setname_np is apparently not available on musl. should i revert or leave the only_for_archs variable?

**Comment by pullmoll** [comment]
You could perhaps shield this non-portable (_np) function call with
```
#if defined(__GLIBC__)
    pthread_setname_np(pthread_self(), name);
#endif
```
i.e. create a patch like `srcpkgs/obs/patches/musl-no_pthread_setname_np.patch` and add these two lines to `libobs/util/threading-posix.c`.

Of course only if this functionality is not essential for the package to work right.

**Comment by lemmi** [comment]
this is more or less ripping out everything none glibc/musl related and replacing it with pullmoll's suggestion. this doesn't change the glibc version and lets it build with musl, too. i can't confirm this is a working fix for musl, though.

**Comment by beefcurtains** [comment]
When writing a patch, try to create only minimal delta. For example:
```
@@ -267,7 +267,7 @@
 	pthread_setname_np(name);
 #elif defined(__FreeBSD__)
 	pthread_set_name_np(pthread_self(), name);
-#elif !defined(__MINGW32__)
+#elif !defined(__MINGW32__) && defined(__GLIBC__)
 	pthread_setname_np(pthread_self(), name);
 #endif
 }
```

File: srcpkgs/obs/patches/0001-musl.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ void os_set_thread_name(const char *name)
+ {
+-#if defined(__APPLE__)
+-	pthread_setname_np(name);
+-#elif defined(__FreeBSD__)
+-	pthread_set_name_np(pthread_self(), name);
+-#elif !defined(__MINGW32__)
+-	pthread_setname_np(pthread_self(), name);
++#if defined(__GLIBC__)
++    pthread_setname_np(pthread_self(), name);
```
Comment (xtraeme) [inline srcpkgs/obs/patches/0001-musl.patch]: I guess this patch won't be accepted by upstream, you've removed the existing cases...

**Comment by lemmi** [comment]
after talking with some other people about this i posted that upstream https://github.com/jp9000/obs-studio/pull/441
i could change the patch to the same thing, but it won't be of any consequences for us. either this gets accepted upstream and then i have to delete this patch again, or it won't then we don't care about the other stuff

**Pull request merged**